test/jdk/java/awt/Paint/ButtonRepaint.java

```diff
@@ -34,15 +34,21 @@ public final class ButtonRepaint extends Button {
 
     public static void main(final String[] args) {
         for (int i = 0; i < 10; ++i) {
-            final Frame frame = new Frame();
-            frame.setSize(300, 300);
-            frame.setLocationRelativeTo(null);
-            ButtonRepaint button = new ButtonRepaint();
-            frame.add(button);
-            frame.setVisible(true);
-            sleep();
-            button.test();
-            frame.dispose();
+            Frame frame = null;
+            try {
+                frame = new Frame();
+                frame.setSize(300, 300);
+                frame.setLocationRelativeTo(null);
+                ButtonRepaint button = new ButtonRepaint();
+                frame.add(button);
+                frame.setVisible(true);
+                sleep();
+                button.test();
+            } finally {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            }
```

Comment:
Would it be cleaner this way?
```suggestion
            Frame frame = new Frame();
            try {
                frame.setSize(300, 300);
                frame.setLocationRelativeTo(null);
                ButtonRepaint button = new ButtonRepaint();
                frame.add(button);
                frame.setVisible(true);
                sleep();
                button.test();
            } finally {
                frame.dispose();
            }
```
This avoids the `null`-check in the `finally` block.